### ISSUE TITLE
[Accessiblity] Narrator cannot announce items correctly when switching the items by using up/down keyboard arrow that without expand the dropdown list panel

### ISSUE THREAD

**Issue by MelonWang1** [dotnet/winforms#12421]
### .NET version

.NET SDK 10.0.100-alpha.1.24553.3

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, this issue cannot repro in .NET 3.1, can repro in .NET 5.0.

.NET 3.1:
![Image](https://github.com/user-attachments/assets/3cd87bea-659a-445b-b475-8468be30fe5b)

### Issue description

When switching the items by using up/down keyboard arrow that without expand the dropdown list panel, Narrator announced the FlatStyle/BoserStyle property item twice, the first time unread in its entirety.
![Image](https://github.com/user-attachments/assets/2c860186-92be-4cac-a191-6959995cedbd)

https://github.com/user-attachments/assets/d30d1426-f17c-4922-81fc-3cd33c0f14f9

**Expected behavior:**
Compare with NVDA tool, Narrator should be read the FlatStyle/BoserStyle property item only once and in its entirety.
![Image](https://github.com/user-attachments/assets/3f29b2b9-88c0-4a73-9268-48dde318729e)

https://github.com/user-attachments/assets/6627cef4-ac48-4c10-a48b-8d1b938d0d2b

### Steps to reproduce

1. Create a .NET Core Winforms project.
2. Add a Label and PropertyGrid controls in form designer. 
3. Set the SelectedObject property to Label1 in propertyGrid1 control. 
4. Build and run project.
5. Open Narrator tool and select FlatStyle/BoserStyle property in propertyGrid control.
6. Use keyboard up and down key to select property item.

**More info:**
1. The issue cannot reproduce on standard combobox control.
2. The issue also can reproduce with other properties.
3. This issue cannot repro in NVDA tool.

**Comment by Tanya-Solyanik** [comment]
@LeafShi1  - we can't backport the fix for the drop down closing only on ENTER until this issue is fixed  as well.

**Comment by LeafShi1** [comment]
For the issue **Narrator announced the FlatStyle/BoserStyle property item twice**, I don't think this is a problem.
The reason for this is that the option is highlighted after it is selected.

- The first announce means that the "Standard" item is selected
- The second announce means that the "Standard" item in the editTextBox is highlighted

The problem is obvious from the announce of the `Cursor` property.
![Image](https://github.com/user-attachments/assets/9ab541a1-d442-41b7-92c8-5a4fc304fa3d)


If we want to solve this problem, we can only make the item selected without selecting it in the edit text box, the result is like below
![Image](https://github.com/user-attachments/assets/5b3b24e2-91a0-4a2c-8b14-b8a11a827e0b)

For issue **the first time unread in its entirety**, it should be fixed in PR #12431, once the edit text box gets the focus, the value can be read out completely.

**Comment by Tanya-Solyanik** [comment]
@LeafShi1  - I added some comments in the PR.

**Comment by Philip-Wang01** [comment]
Verified this issue in the latest .NET 10.0 SDK build: 10.0.100-alpha.1.24611.5 + binaries built from the main branch of Winforms repo, it was fixed: When switching the items by using up/down keyboard arrow that without expand the dropdown list panel, Narrator announced the property item entirety.
![Image](https://github.com/user-attachments/assets/52941647-0b05-46b1-af58-6b1279ed05ba)
